eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/MultiReaderUnifiedSet.java

```diff
@@ -426,6 +426,16 @@ public MutableShortSet collectShort(ShortFunction<? super T> shortFunction)
     @Override
     public boolean equals(Object o)
     {
+        if (o == this || o == this.delegate)
+        {
+            return true;
+        }
+
+        if (!(o instanceof Set))
+        {
+            return false;
+        }
```

Comment:
This one is worth keeping because it avoids grabbing the lock. Is it covered by tests?

Reply:
I'm not sure, but I updated that test because it wasn't passing anymore with the change: `MultiReaderUnifiedSetScalaTest`